src/main/java/com/synopsys/integration/detect/interactive/CliDecisionBranch.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * synopsys-detect
+ *
+ * Copyright (c) 2020 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.detect.interactive;
+
+import com.synopsys.integration.detect.configuration.DetectProperties;
+
+public class CliDecisionBranch implements DecisionTree {
+    private final boolean connectedToBlackDuck;
+
+    public CliDecisionBranch(boolean connectedToBlackDuck) {
+        this.connectedToBlackDuck = connectedToBlackDuck;
+    }
+
+    public void traverse(InteractivePropertySourceBuilder propertySourceBuilder, InteractiveWriter writer) {
+        if (connectedToBlackDuck) {
+            Boolean upload = writer.askYesOrNo("Would you like to upload CLI scan results to the Black Duck server?");
```

Comment:
This also, I assume, has nothing to do with this PR, but instead of "CLI scan", I believe we usually use "signature scan".

Reply:
Very good, I'll change that now.